internal/rbd/rbd_util.go

```diff
@@ -843,62 +939,79 @@ func (rv *rbdVolume) getImageInfo() error {
 	}
 	fs := librbd.FeatureSet(features)
 	rv.ImageFeatures = strings.Join(fs.Names(), ",")
+	rv.ParentName = imageInfo.Parent_name
 
 	return nil
 }
 
-// snapInfo strongly typed JSON spec for snap ls rbd output
-type snapInfo struct {
-	ID        int64  `json:"id"`
-	Name      string `json:"name"`
-	Size      int64  `json:"size"`
-	Timestamp string `json:"timestamp"`
+func (rv *rbdVolume) getCloneDepth(ctx context.Context) (int, error) {
+	depth := 0
+	imageName := rv.RbdImageName
+	poolName := rv.Pool
+	for {
+		if imageName == "" {
+			return depth, nil
+		}
+		vol := rbdVolume{
+			Pool:         poolName,
+			RbdImageName: imageName,
+		}
+		err := vol.getImageInfo()
```

Comment:
This most likely connects the rbdVolume, so it needs to be destroyed afterwards. Easiest is to always add a `defer vol.Destroy()` immediately after creating the rbdVolume object.

Reply:
will make changes to reuse the ioctx and conn of parent volume, no need to destroy it, as parent will destroy it.

Reply:
try not to over complicate things, if the connection exists. Optimizations like this can be done later when ioctx's are in a shared pool.

Reply:
we cannot use getImageInfo as it doesn't return the parent Image name, filed an issue https://github.com/ceph/go-ceph/issues/312. in followup PR i will change back to getImageInfo.